src/bun.js/node/node_fs_watcher.zig

```diff
@@ -403,6 +405,31 @@ pub const FSWatcher = struct {
 
                             current_task.append(relative_path, event_type, .destroy);
 
+                            _on_file_update_path_buf[path_slice.len] = 0;
+                            const full_path = _on_file_update_path_buf[0..path_slice.len :0];
+
+                            var is_watched = false;
+                            if (this.file_paths.contains(full_path)) {
+                                is_watched = true;
+                            }
+                            // Add new files or directories to the watch list.
+                            if (!is_watched) {
+                                const full_path_clone = bun.default_allocator.dupe(u8, full_path) catch unreachable;
+                                this.file_paths.put(full_path_clone, {}) catch unreachable;
+                                const fs_info = fdFromAbsolutePathZ(full_path) catch null;
+                                if (fs_info) |info| {
+                                    if (info.is_file) {
+                                        this.default_watcher.?.addFileAssumeLocked(info.fd, full_path_clone, file_hash, options.Loader.file, 0, null, false) catch unreachable;
+                                        const relative_path_clone = bun.default_allocator.dupe(u8, relative_path) catch unreachable;
```

Comment:
If we save the file paths and their relative paths into a HashMap we can avoid recreating the relative and dupe it here, and just do current_task with `.none` so we only clone it once

Reply:
What was the original purpose of the `file_paths` variable? I haven't seen any place in the code where the values from `file_paths` are read (except for this pr). Is this s used to store some path variables that need to be freed by this struct? Should I define `file_paths` as a structure with keys as `full_path` and values as `relative_path` to better represent the association between paths?


Thanks.

Reply:
The original purpose is to keep it alive so the watcher.zig can use it safely (it expect that this paths are alive), but at this point we can use it to avoid the extra dupes. key as full path and values as relative sounds good for this case.

Reply:
Fixed.